### PR TITLE
[4.0] Clean up JHtmlBehavior

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue # .

### Summary of Changes

- Remove deprecated code
- don't use inline script for caption
- don't use inline script for multiselect
- don't use inline script for highlighter

### Documentation Changes Required
??

@wilsonge please merge this one (if you agree with the removals, else let me know and I will revert the ones that are not ok) so the GSOC student can start implementing the javascript test

### CODE REVIEW

File: media/system/js/highlighter.js [hunk 1]
```diff
@@ -4,108 +4,128 @@
...
-            }
-        }
-        return 0;
-    },
+		    recurse = function (node, regex, klass) {
+			    if (node.nodeType === 3) {
+				    var match = node.nodeValue.match(regex), highlight, $highlight, wordNode, wordClone, comparer, i;
+				    if (match) {
+					    highlight = document.createElement(options.tag);
+					    $highlight = highlight;
```
Comment (C-Lodder) [inline media/system/js/highlighter.js]: `$highlight = highlight;`. Don't needs this

File: media/system/js/highlighter.js [hunk 2]
```diff
@@ -4,108 +4,128 @@
...
+				return true;
+			}
+			highlighter = new Joomla.Highlighter({
+				startElement: start,
+				endElement: end,
+				className: options['class'],
+				onlyWords: false,
+				tag: options.tag
+			}).highlight(options.terms);
+			document.querySelector(start).remove();
```
Comment (C-Lodder) [inline media/system/js/highlighter.js]: IE has no support for `remove()`. Best use something like:

```
var startEl = document.querySelector(start);
startEl.parentNode.removeChild(startEl);
```

File: media/system/js/highlighter.js (hunk 1 above)
Comment (dgt41) [inline media/system/js/highlighter.js]: I did the jQuery to vanilla conversion just by reading the code, probably it's totally broken, but it's ok as the GSOC student will fix it as he'll be writing the Javascript tests

File: media/system/js/highlighter.js (hunk 2 above)
Comment (dgt41) [inline media/system/js/highlighter.js]: IE...

Comment (C-Lodder) [inline media/system/js/highlighter.js]: `$(start).remove();`  :satisfied:

File: media/system/js/highlighter.js [hunk 3]
```diff
@@ -4,108 +4,128 @@
...
-                return 1;
-            }
-        } else if ((node.nodeType === 1 && node.childNodes) && !/(script|style|textarea|iframe)/i.test(node.tagName) && !(node.tagName === options.tag.toUpperCase() && node.className === klass)) {
-            for (i = 0; i < node.childNodes.length; i++) {
-                i += recurse(node.childNodes[i], regex, klass);
-            }
-        }
-        return 0;
-    },
+		    recurse = function (node, regex, klass) {
```
Comment (brianteeman) [inline media/system/js/highlighter.js]: is this supposed to be K?

Comment (C-Lodder) [inline media/system/js/highlighter.js]: yup

Comment (dgt41) [inline media/system/js/highlighter.js]: class and className are reserved words, cannot be used as  vars

Comment (brianteeman) [inline media/system/js/highlighter.js]: thanks for the explanation

**Pull request closed without merging**